imagedephi/redact/tiff.py

```diff
@@ -82,26 +82,46 @@ def _iter_tiff_tag_entries(
     def __init__(
         self,
         tiff_info: TiffInfo,
-        base_rule_set: RuleSet,
-        override_rule_set: RuleSet | None = None,
+        base_rule_set: Ruleset,
+        override_rule_set: Ruleset | None = None,
     ) -> None:
         self.tiff_info = tiff_info
 
         self.redaction_steps = {}
         self.no_match_tags = []
         ifds = self.tiff_info["ifds"]
-        override_rules = override_rule_set.get_metadata_tiff_rules() if override_rule_set else []
-        base_rules = base_rule_set.get_metadata_tiff_rules()
+        override_rules = (
+            override_rule_set.get_format_rules(self.file_format).metadata
+            if override_rule_set
+            else None
+        )
+        base_rules: dict[str, ConcreteMetadataRule] = base_rule_set.get_format_rules(
+            self.file_format
+        ).metadata
+        merged_rules = base_rules | override_rules if override_rules else base_rules
         for tag, _ in self._iter_tiff_tag_entries(ifds):
-            # First iterate through overrides, then base
-            for rule in chain(override_rules, base_rules):
-                if rule.is_match(tag):
-                    self.redaction_steps[tag.value] = rule
+            tag_rule = None
+            for name in [tag.name] + list(tag.get("altnames", set())):
+                tag_rule = merged_rules.get(name, None)
+                if tag_rule and self.is_match(tag_rule, tag):
+                    self.redaction_steps[tag.value] = tag_rule
                     break
             else:
-                # End of iteration, without "break"; no matching rule found anywhere
                 self.no_match_tags.append(tag)
 
+    def is_match(self, rule: ConcreteMetadataRule, tag: tifftools.TiffTag) -> bool:
+        if rule.action in ["keep", "delete", "replace"]:
+            rule_tag = get_tiff_tag(rule.key_name)
+            return rule_tag.value == tag.value
+        return False
+
+    def apply(self, rule: ConcreteMetadataRule, ifd: IFD) -> None:
+        tag = get_tiff_tag(rule.key_name)
+        if rule.action == "delete":
+            del ifd["tags"][tag.value]
+        elif rule.action == "replace":
+            ifd["tags"][tag.value]["data"] = rule.new_value
```

Comment:
Note that this functionality has been moved from the old `Rule` classes to the `RedactionPlan` classes. If this change feels like the concept of a `RedactionPlan` is becoming bloated, I'd like to discuss potential designs that separate these concerns better.